openwebstart/src/main/java/com/openwebstart/jvm/JavaRuntimeSelector.java

```diff
@@ -37,10 +39,20 @@ public static void setAskForUpdateFunction(final Predicate<RemoteJavaRuntime> as
         getInstance().askForUpdateFunction = askForUpdateFunction;
     }
 
-    public LocalJavaRuntime getRuntime(final VersionString versionString, final String vendor, final URI serverEndpoint) throws Exception {
+    @Override
+    public Path getJavaHome(VersionString version, URL url) {
+        try {
+            return getRuntime(version, null, url).getJavaHome();
+        } catch (Exception e) {
+            LOG.info("Exception while getting runtime - " + version + " - " + url, e);
+            return null;
+        }
+    }
+
+    public LocalJavaRuntime getRuntime(final VersionString versionString, final String vendor, final URL serverEndpoint) throws Exception {
         Assert.requireNonNull(versionString, "versionString");
 
-        LOG.debug("Trying to find Java runtime. Requested version: '" + versionString + "' Requested vendor: '" + vendor);
+        LOG.debug("Trying to find Java runtime. Requested version: '" + versionString + "' Requested vendor: '" + vendor + "'");
```

Comment:
consider to use {} instead of string concatenation for consistency to other log statements in this PR and better readability here